planner/core/logical_plan_builder.go

```diff
@@ -2112,11 +2113,26 @@ func (b *PlanBuilder) BuildDataSourceFromView(dbName model.CIStr, tableInfo *mod
 	if err != nil {
 		return nil, err
 	}
+
+	originalVisitInfo := b.visitInfo
+	b.visitInfo = make([]visitInfo, 0)
 	selectLogicalPlan, err := b.Build(selectNode)
 	if err != nil {
 		return nil, err
 	}
 
+	if tableInfo.View.Security == model.SecurityDefiner {
+		if pm := privilege.GetPrivilegeManager(b.ctx); pm != nil {
+			for _, v := range b.visitInfo {
+				if !pm.RequestVerificationWithUser(v.db, v.table, v.column, v.privilege, tableInfo.View.Definer) {
```

Comment:
OK，this way seems work well.

Reply:
@tiancaiamao This way do not work, otherwise I create a new session and a new context with `Definer`, that will make like there is a temporary use login to TiDB.

Reply:
OK, I'll have a refactor later after it's merged.